hdbscan/_hdbscan_tree.pyx

```diff
@@ -725,8 +725,10 @@ cpdef tuple get_clusters(np.ndarray tree, dict stability,
     # if you do, change this accordingly!
     if allow_single_cluster:
         node_list = sorted(stability.keys(), reverse=True)
+        node_list = [int(n) for n in node_list]
     else:
         node_list = sorted(stability.keys(), reverse=True)[:-1]
+        node_list = [int(n) for n in node_list]
```

Comment:
I don't like it, but it works.  Cython and NumPy have a weird interaction here with allowing conversion from float or not depending on both Cython and NumPy versions.

If you want to convert from float, you could also just `cdef` it to `long` or `int`, IIRC that should always work, and I don't think you care about mroe than 2**31 clusters....

Reply:
I do agree that there are some weird interactions here. It is down to the fact that at some point the values are contained in a single numpy array of type float64 (since other values in the array are floats) and this was "fine", but it is causing catches now in the cython/numpy interactions. I think this workaround looks fine under the circumstances.